tests/Directory.Build.targets

```diff
@@ -35,7 +35,7 @@
 
   <!-- Used for running one helix job per test class -->
   <Target Name="_ExtractTestClassNames"
-          Condition="'$(ExtractTestClassNamesForHelix)' == 'true'"
+          Condition="'$(ExtractTestClassNamesForHelix)' == 'true' and '$(ArchiveTests)' == 'true'"
```

Comment:
Why are we building anything in the `tests` directory in this build configuration? If we don't want to do "test things", do we need to build the test assemblies?

Reply:
That's a good question. AFAICS, even if we have `-p:Tests=false` like used for the codeql build, it still builds the test assemblies because they are referenced from the sln.

Reply:
I didn't think it uses the sln, but instead uses:

https://github.com/dotnet/aspire/blob/a8b0bed9bcf0e09ca6274e08b2ecdc145ae6df81/eng/Build.props#L2-L8

Reply:
Oh then we should definitely make building the tests conditional. I'll open an issue to track that work.